File: src/main/java/com/github/caijh/commons/util/DesUtils.java
package com.github.caijh.commons.util;

import java.nio.charset.StandardCharsets;
import java.security.Key;
import javax.crypto.spec.SecretKeySpec;

public class DesUtils implements SecretUtils {

    private static final DesUtils INSTANCE = new DesUtils();

    private DesUtils() {

    }

    public static DesUtils getInstance() {
        return INSTANCE;
    }

    @Override
    public String keyAlgorithm() {
        return "DES";
    }

    @Override
    public String cipherAlgorithm() {
        return "DES";
    }

    @Override
    public Key secretKey(String key) {
        byte[] result = new byte[8];
        byte[] keys = key.getBytes(StandardCharsets.UTF_8);
        for (int i = 0; i < 8; i++) {
            if (i < keys.length) {
                result[i] = keys[i];
            } else {
                result[i] = 0x01;
            }
        }
        return new SecretKeySpec(result, keyAlgorithm());
    }

}

File: src/main/kotlin/com/github/caijh/commons/util/DesUtils.kt
package com.github.caijh.commons.util

import java.nio.charset.StandardCharsets
import java.security.Key
import javax.crypto.spec.SecretKeySpec

class DesUtils private constructor() : SecretUtils {
    override fun keyAlgorithm(): String {
        return "DES"
    }

    override fun cipherAlgorithm(): String {
        return "DES"
    }

    override fun secretKey(key: String): Key {
        val result = ByteArray(8)
        val keys = key.toByteArray(StandardCharsets.UTF_8)
        for (i in 0..7) {
            if (i < keys.size) {
                result[i] = keys[i]
            } else {
                result[i] = 0x01
            }
        }
        return SecretKeySpec(result, keyAlgorithm())
    }

    companion object {
        val instance = DesUtils()
    }
}
